drivers/power/ltc7841/iio_ltc7841.c

```diff
@@ -0,0 +1,500 @@
+/***************************************************************************//**
+ *   @file   iio_ltc7841.c
+ *   @brief  Source file for the LTC7841 IIO Driver
+ *   @author Marvin Cabuenas (marvinneil.cabuenas@analog.com)
+********************************************************************************
+ * Copyright 2024(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+#include "no_os_units.h"
+#include "no_os_util.h"
+#include "iio_ltc7841.h"
+
+static uint8_t rsense_value_used;
+static struct iio_device ltc7841_iio_dev;
+
+enum ltc7841_iio_chan_type {
+	LTC7841_IIO_VOUT_CHAN,
+	LTC7841_IIO_VIN_CHAN,
+	LTC7841_IIO_IOUT_CHAN,
+	LTC7841_IIO_IIN_CHAN,
+	LTC7841_IIO_TEMP_CHAN,
+};
+
+/**
+ * @brief Read register value.
+ * @param dev     - The iio device structure.
+ * @param reg	  - Register to read.
+ * @param readval - Read value.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int ltc7841_iio_reg_read(void *dev, uint32_t reg, uint32_t *readval)
+{
+	uint8_t register_value[2];
+	int ret;
+	struct ltc7841_iio_desc *iio_ltc7841 = dev;
+	struct ltc7841_desc *ltc7841 = iio_ltc7841->ltc7841_desc;
+
+	ret = ltc7841_reg_read(ltc7841, (uint8_t)reg, register_value);
+	*readval = no_os_get_unaligned_le16(register_value);
```

Comment:
i'm not sure what's happening here, readval is a `uint32_t *`, i'm pretty sure on one of the endianesses this will fail